Source/WebCore/platform/graphics/PixelBuffer.h

```diff
@@ -61,6 +61,18 @@ class PixelBuffer {
 
     PixelBuffer deepClone() const;
 
+    uint8_t* bytes() const;
+    size_t byteLength() const;
+
+    bool setRange(const uint8_t* data, size_t dataByteLength, size_t byteOffset);
+    bool zeroRange(size_t byteOffset, size_t rangeByteLength);
+    void zeroFill() { zeroRange(0, byteLength()); }
+
+    uint8_t item(size_t index) const;
+    void set(size_t index, double value);
```

Comment:
I'm confused that the argument is a double, not a uint8_t

Reply:
The caller should be in charge of how values get clamped etc, not PixelBuffer. And is the argument in the 0-255 range, or the 0-1 range?

Reply:
Yes this is right. But the code of the GTK and Windows ports rely on the clamping of Uint8ClampedAdaptor. The code in FEColorMatrixSoftwareApplier::applyPlatformUnaccelerated() does the calculations of the color channel in floating point and it was relying on Uint8ClampedArray to do the clamping once it calls its set() method. I think this can be done in a separate patch.